libkatana_python_native/include/katana/python/PropertyGraphPython.h

```diff
@@ -33,4 +34,16 @@ PythonArgumentsToTable(
 
 }  // namespace katana::python
 
+// Banned types based on the holders used for graph related types.
+
+template <typename T>
+struct ::pybind11::detail::type_caster<std::unique_ptr<katana::EntityIndex<T>>>
+    : public katana::python::banned_type_caster<
+          std::unique_ptr<katana::EntityIndex<T>>> {};
+
+template <>
+struct ::pybind11::detail::type_caster<std::unique_ptr<katana::PropertyGraph>>
+    : public katana::python::banned_type_caster<
+          std::unique_ptr<katana::PropertyGraph>> {};
```

Comment:
This is interesting, so basically if someone tries to create a `std::unique_ptr` reference to PropertyGraph they have to cast it in this way and this struct will throw an error when the cast tries to happen? Do you have a clearer example of incorrect usage? It might help me picture how the code here comes into play.

Reply:
The caster is used by pybind11 when casting objects to/from python objects. This caster causes a *static* assert when this happens.  At least in some cases. There is lengthy documentation in `BanTypes.h`.

This will not affect use of unique_ptr<PropertyGraph> at the C++ level.

Reply:
Gotcha. Thanks for the clarification.